tests/unit/test_session.py

```diff
@@ -250,16 +299,92 @@ class _Session(SparklySession):
+            'PYSPARK_SUBMIT_ARGS': (
+                '--conf "spark.app.id=sparkly-process_uid-456789" '
+                '--conf "my.conf.here=5g" --and-other-properties'
+            ),
         }
 
         SparklySession()
         SparklySession.stop()
 
         self.assertEqual(os_mock.environ, {
-            'PYSPARK_SUBMIT_ARGS': '--conf "my.conf.here=5g" --and-other-properties',
+            'PYSPARK_SUBMIT_ARGS': (
+                '--conf "spark.app.id=sparkly-process_uid-456789" '
+                '--conf "my.conf.here=5g" --and-other-properties'
+            ),
+        })
+
+    @mock.patch('sparkly.session.time.time')
+    @mock.patch('sparkly.session.uuid.uuid4')
+    @mock.patch('sparkly.session.os')
+    def test_session_null_app_id_template(self, os_mock, uid_mock, time_mock):
+        time_mock.return_value = 456789
+        uid_mock.return_value = 'session_uid'
+        os_mock.environ = {}
+
+        class Session(SparklySession):
+            app_id_template = None
+
+        Session()
+
+        self.assertEqual(os_mock.environ, {
+            'PYSPARK_PYTHON': sys.executable,
+            'PYSPARK_SUBMIT_ARGS': (
+                # MISSING: '--conf "spark.app.id=sparkly-process_uid-456789" '
+                '--conf "spark.sql.catalogImplementation=hive" '
+                'pyspark-shell'
+            ),
+        })
+
+    @mock.patch('sparkly.session.time.time')
+    @mock.patch('sparkly.session.uuid.uuid4')
+    @mock.patch('sparkly.session.os')
+    def test_session_empty_app_id_template(self, os_mock, uid_mock, time_mock):
+        time_mock.return_value = 456789
+        uid_mock.return_value = 'session_uid'
+        os_mock.environ = {}
+
+        class Session(SparklySession):
+            app_id_template = ''
+
+        Session()
+
+        self.assertEqual(os_mock.environ, {
+            'PYSPARK_PYTHON': sys.executable,
+            'PYSPARK_SUBMIT_ARGS': (
+                # MISSING: '--conf "spark.app.id=sparkly-process_uid-456789" '
+                '--conf "spark.sql.catalogImplementation=hive" '
+                'pyspark-shell'
+            ),
+        })
+
+    @mock.patch('sparkly.session.time.time')
+    @mock.patch('sparkly.session.uuid.uuid4')
+    @mock.patch('sparkly.session.os')
+    def test_session_custom_app_id_template(self, os_mock, uid_mock, time_mock):
+        time_mock.return_value = 456789
+        uid_mock.return_value = 'session_uid'
+        os_mock.environ = {}
+
+        class Session(SparklySession):
+            app_id_template = 'my_custom-{process_uid}-{session_uid}-{process_time}-{session_time}'
+
+        Session()
+
+        self.assertEqual(os_mock.environ, {
+            'PYSPARK_PYTHON': sys.executable,
+            'PYSPARK_SUBMIT_ARGS': (
+                '--conf "spark.app.id=my_custom-process_uid-session_uid-123456-456789" '
+                '--conf "spark.sql.catalogImplementation=hive" '
+                'pyspark-shell'
+            ),
```

Comment:
Should we also add a test demoing that an explicitly set `spark.app.id` "overrides" the template?